avocado/utils/download.py

```diff
@@ -87,30 +87,33 @@ def url_download_interactive(url, output_file, title='', chunk_size=102400):
     """
     output_dir = os.path.dirname(output_file)
     output_file = open(output_file, 'w+b')
-    input_file = urllib2.urlopen(url)
 
     try:
+        input_file = urllib2.urlopen(url)
         file_size = int(input_file.headers['Content-Length'])
+
+        logging.info('Downloading %s, %s to %s', os.path.basename(url),
+                     output.display_data_size(file_size), output_dir)
+
+        progress_bar = output.ProgressBar(maximum=file_size, title=title)
+
+        # Download the file, while interactively updating the progress
+        progress_bar.draw()
+        while True:
+            data = input_file.read(chunk_size)
+            if data:
+                progress_bar.append_amount(len(data))
+                output_file.write(data)
+            else:
+                progress_bar.update_amount(file_size)
+                break
     except KeyError:
         raise ValueError('Could not find file size in HTTP headers')
-
-    logging.info('Downloading %s, %s to %s', os.path.basename(url),
-                 output.display_data_size(file_size), output_dir)
-
-    progress_bar = output.ProgressBar(maximum=file_size, title=title)
-
-    # Download the file, while interactively updating the progress
-    progress_bar.draw()
-    while True:
-        data = input_file.read(chunk_size)
-        if data:
-            progress_bar.append_amount(len(data))
-            output_file.write(data)
-        else:
-            progress_bar.update_amount(file_size)
-            break
-
-    output_file.close()
+    except Exception as e:
+        logging.warn('Please, check your internet connection: %s',
```

Comment:
Well if it fails to download the file, then it should IMO crash. In case of interactive download it might as for skip/repeat, but in general it failed, so it should report the failure and not just log a warning and proceed.

Btw in vt-bootstrap you can use `--no-downloads` and avoid the download when you're outside the connectivity.